pkg/hubble/filters/fqdn.go

```diff
@@ -88,7 +88,7 @@ func filterByFQDNs(fqdnPatterns []string, getFQDNs func(*v1.Event) []string) (Fi
 // The filter function returns true if and only if the DNS query field matches any of
 // the regular expressions.
 func filterByDNSQueries(queryPatterns []string) (FilterFunc, error) {
-	var queries []*regexp.Regexp
+	queries := make([]*regexp.Regexp, 0, len(queryPatterns))
```

Comment:
With this type of pre-allocation, you should consider using an index notation to get more of a performance benefit.

Example:

```
❯ cat main_test.go
package main

import (
        "bytes"
        "io"
        "testing"
)

const (
        l = 50
)

type stuff struct {
        idx int

        // bunch of random stuff
        io.Writer
        bytes.Buffer
        a, b, c int
        x, y, z [4]byte
}

func BenchmarkAppend(b *testing.B) {
        for i := 0; i < b.N; i++ {
                s := make([]stuff, 0, l)
                for j := 0; j < l; j++ {
                        s = append(s, stuff{idx: j})
                }
        }
}

func BenchmarkIndex(b *testing.B) {
        for i := 0; i < b.N; i++ {
                s := make([]stuff, l, l)
                for j := 0; j < l; j++ {
                        s[j] = stuff{idx: j}
                }
        }
}

```

```
❯ go test -bench=. -v
goos: darwin
goarch: amd64
BenchmarkAppend
BenchmarkAppend-12       2671011               438 ns/op
BenchmarkIndex
BenchmarkIndex-12       88476345                13.1 ns/op
PASS
ok      _/Users/glibsm/code/tmp/prealloc        2.884s
```

Reply:
Wow nice! I wasn't aware that there is such a large difference in terms of performance. Will address this in a follow-up PR, potentially covering other cases previously converted.